### ISSUE TITLE
What's the reason @types are being added as dependencies as opposed to peerDependencies?

### ISSUE THREAD

**Issue by gajus** [testing-library/react-hooks-testing-library#729]
https://github.com/testing-library/react-hooks-testing-library/blob/ca13ed9ae8b9c7dc92ffe586e48cbf46f1cfc3a9/package.json#L53-L55

This is causing multiple React declarations errors in our codebase, e.g.

```
node_modules/@testing-library/react-hooks/node_modules/@types/react/index.d.ts:3127:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'datalist' must be of type 'DetailedHTMLProps<HTMLAttributes<HTMLDataListElement>, HTMLDataListElement>', but here has type 'DetailedHTMLProps<HTMLAttributes<HTMLDataListElement>, HTMLDataListElement>'.

3127             datalist: React.DetailedHTMLProps<React.HTMLAttributes<HTMLDataListElement>, HTMLDataListElement>;
                 ~~~~~~~~

  node_modules/@types/react/index.d.ts:3126:13
    3126             datalist: React.DetailedHTMLProps<React.HTMLAttributes<HTMLDataListElement>, HTMLDataListElement>;
                     ~~~~~~~~
    'datalist' was also declared here.
```

**Comment by ksjogo** [comment]
Hey, we are currently looking into adding this library into our codebase and the functionality is great.
But we were surprised that the library was pulling in @types as dependencies. Its a flow codebase, so they aren't already present, but @testing-library/react-hooks would be the only dependent.

From previous experiences with other ts libraries in our codebase, we would have expected the types to be devDependencies.

**Comment by mpeyper** [comment]
Hi @ksjogo.  The main reason we use a `dependency` instead of a `peerDependency` is for folks like yourself, who do not want to install the types to silence the `peerDependency` warning NPM will spit out at you if you don't have it installed.

We are a typescript library and as such our TS users expect our dependencies to be installed and as they form part of our public API then `dependency` is, in my opinion, the correct place for them and, in general, installing the types when they are not needed is not an issue.

That said, I'm strongly considering moving them into `peerDependency` and marking them as `peerDependencyMeta` `optional` as the vast majority of out users will already have the types installed anyway, and it would fix the `yarn` issue (#610) and stop this question from being asked so often.

**Comment by mpeyper** [comment]
This should be resolved in v8.0.0 where we no longer install type dependencies.
